_posts/2019-07-15-mosquitto-mqtt.md

```diff
@@ -0,0 +1,185 @@
+---
+layout: post
+current: post
+cover: Image_5.png
+title: Using WoTT to Cryptographically Secure access between a Mosquitto Brokered MQTT Client and Server
+date: 2019-07-15 17:00:00
+category: tutorials
+author: Fiona McAllister
+tags: [Open Source]
+class: post-template
+---
+
+# Using WoTT to Secure Access to a Mosquitto MQTT Server
+
+## Introduction 
+
+Mosquitto is a lightweight message broker for MQTT. MQTT itself is a pub-sub messaging protocol that is particularly popular amongst IoT applications due to its optimisation for high-latency networks. Together, they are effective for IoT usage being low-power to run and adding a layer of security to IoT devices. 
+By default the connection between a Mosquitto run client and server is unencrypted. Mosquitto does have provisions for securing messages between client and server that are very easy to configure- though most of these options involve a password file.
+
+Here we'll look at using WoTT's unique device ID to authenticate and secure access for a Mosquitto brokered MQTT protocol essentially identifying the devices cryptographically removing the need for usernames and passwords. This example looks at publishing messages to a temperature topic hosted by the Mosquitto server.
+
+You will need 2 WoTT devices with the [WoTT Agent]({site.url}/documentation/getting-started) installed. We reccommend that at least one of these is a Debian machine for the server (although 2 Pis is fine).
```

Comment:
Rule of thumb (at least in the US), is to always write out numbers less than 10 (https://www.writersdigest.com/online-editor/when-do-i-spell-out-numbers).

Reply:
"reccommend" -> "recommend"

Reply:
oh also `{site.url}` -> `{{site.url}}`